django_comments/views/comments.py

```diff
@@ -27,105 +33,157 @@ def __init__(self, why):
+            if response is False:
+                raise BadRequest("comment_will_be_posted receiver %r killed the comment" % receiver.__name__)
+
+        # Save the comment and signal that it was saved
+        comment.save()
+        signals.comment_was_posted.send(
+            sender=comment.__class__,
+            comment=comment,
+            request=self.request
+        )
+        return comment
+
+    def get_template_names(self):
+        if self.template_name is None:
+            model = type(self.target_object)
+            return [
+                # These first two exist for purely historical reasons.
+                # Django v1.0 and v1.1 allowed the underscore format for
+                # preview templates, so we have to preserve that format.
+                "comments/%s_%s_preview.html" % (model._meta.app_label, model._meta.model_name),
+                "comments/%s_preview.html" % model._meta.app_label,
+                # Now the usual directory based template hierarchy.
+                "comments/%s/%s/preview.html" % (model._meta.app_label, model._meta.model_name),
+                "comments/%s/preview.html" % model._meta.app_label,
+                "comments/preview.html",
+            ]
+        else:
+            return [self.template_name]
+
+    def get_context_data(self, form):
+        return dict(
+            form=form,
+            comment=form.data.get("comment", ""),
+            next=self.data.get("next", self.kwargs.get('next')),
+        )
+
+    @method_decorator(csrf_protect)
+    def dispatch(self, *args, **kwargs):
+        return super().dispatch(*args, **kwargs)
+
+    def post(self, request, **kwargs):
+        self.object = None
+        self.target_object = None
+        self.data = self.get_form_kwargs()
+        try:
+            self.target_object = self.get_target_object(self.data)
+        except BadRequest as exc:
+            return exc.response
+
+        form = self.get_form()
+
+        # Check security information
+        if form.security_errors():
             return CommentPostBadRequest(
-                "comment_will_be_posted receiver %r killed the comment" % receiver.__name__)
-
-    # Save the comment and signal that it was saved
-    comment.save()
-    signals.comment_was_posted.send(
-        sender=comment.__class__,
-        comment=comment,
-        request=request
-    )
-
-    return next_redirect(request, fallback=next or 'comments-comment-done',
-                         c=comment._get_pk_val())
+                "The comment form failed security verification: %s" % escape(str(form.security_errors())))
+
+        if not form.is_valid() or "preview" in self.data:
+            return self.form_invalid(form)
+        else:
+            try:
+                self.object = self.create_comment(form)
+            except BadRequest as exc:
+                return exc.response
+            else:
+                return self.form_valid(form)
```

Comment:
Similar comment. FormView is supposed to be doing this for you and you can simply override form_valid/form_invalid methods.

Reply:
It might be a question of taste..